MaxText/layers/attentions.py

```diff
@@ -632,6 +634,12 @@ def cudnn_flash_attention(
       mask_type = "padding_causal"  # only padding_causal mask type can take a created mask
       attn_mask = self.generate_attention_mask(query, key, decoder_segment_ids, model_mode)
 
+    # scaling factor for fused query weight or not
+    if self.scale_factor is None:
+      scale_factor = 1.0 / jnp.sqrt(query.shape[-1])
+    else:
+      scale_factor = self.scale_factor
+
```

Comment:
same comment as above, let's just use 1 to be consistent with the TPU and DotProduct attention.